platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/UserPropertiesDeletionScheduler.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.insecure.defaults.service;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.codice.ddf.configuration.AbsolutePathResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UserPropertiesDeletionScheduler {
+
+  @VisibleForTesting
+  static final Path TEMP_TIMESTAMP_FILE_PATH =
+      Paths.get(new AbsolutePathResolver("data/tmp/timestamp.bin").getPath());
+
+  @VisibleForTesting
+  static final Path USERS_PROPERTIES_FILE_PATH =
+      Paths.get(new AbsolutePathResolver("etc/users.properties").getPath());
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(UserPropertiesDeletionScheduler.class);
+  private static final String ROUTE_ID = "deletionJob";
+  private final CamelContext context;
+  private boolean usersPropertiesDeletionFlag;
+
+  public UserPropertiesDeletionScheduler() {
+    context = new DefaultCamelContext();
+  }
+
+  public void scheduleDeletion() {
+    if (usersPropertiesDeletionFlag && USERS_PROPERTIES_FILE_PATH.toFile().exists()) {
+      String cron = getCron();
+      if (cron == null) {
+        return;
+      }
+      try {
+        RouteBuilder builder =
+            new RouteBuilder() {
+              @Override
+              public void configure() throws Exception {
+                fromF("quartz2://scheduler/deletionTimer?cron=%s", cron)
+                    .routeId(ROUTE_ID)
+                    .bean(UserPropertiesDeletionScheduler.class, "deleteFiles");
+              }
+            };
+        context.addRoutes(builder);
+        context.start();
+      } catch (Exception e) {
+        LOGGER.debug("Unable to create camel route.", e);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  String getCron() {
```

Comment:
✏️  Would be nice if we could hide this from test.

Reply:
I think this method is important to be visible for testing because we can test the cron calculation and file creation without scheduling any tasks. That will help us to see what happens in different scenarios. I did try to minimize what I expose for testing in other parts though.